pkg/server/registration/manager.go

```diff
@@ -0,0 +1,91 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	telemetry_server "github.com/spiffe/spire/pkg/common/telemetry/server"
+	"github.com/spiffe/spire/pkg/common/util"
+	"github.com/spiffe/spire/proto/spire/server/datastore"
+)
+
+const (
+	_pruningCandence = 2 * time.Hour
+)
+
+// ManagerConfig is the config for the registration manager
+type ManagerConfig struct {
+	DataStore datastore.DataStore
+
+	Log     logrus.FieldLogger
+	Metrics telemetry.Metrics
+
+	Clock clock.Clock
+}
+
+// Manager is the manager of registrations
+type Manager struct {
+	c       ManagerConfig
+	log     logrus.FieldLogger
+	metrics telemetry.Metrics
+}
+
+// NewManager creates a new registration manager
+func NewManager(c ManagerConfig) *Manager {
+	if c.Clock == nil {
+		c.Clock = clock.New()
+	}
+
+	return &Manager{
+		c:       c,
+		log:     c.Log.WithField(telemetry.RetryInterval, _pruningCandence),
+		metrics: c.Metrics,
+	}
+}
+
+// Run runs the registration manager
+func (m *Manager) Run(ctx context.Context) error {
+	err := util.RunTasks(ctx,
```

Comment:
This method is already called with a RunTasks from server.go so we can probably just make this function `return pruneEvery(ctx)`